src/nvim/highlight_group.c

```diff
@@ -231,6 +231,7 @@ static const char *highlight_init_both[] = {
   "default link DiagnosticSignOk DiagnosticOk",
   "default DiagnosticDeprecated cterm=strikethrough gui=strikethrough guisp=Red",
   "default link DiagnosticUnnecessary Comment",
+  "default link LspInlayHint Special",
```

Comment:
`Special` might be too colorful for inlay hint? Not sure about other colorschemes, but on `gruvbox` it's orange.

<img width="406" alt="image" src="https://github.com/neovim/neovim/assets/8050659/fe013087-ed49-4ecb-8eb5-ea2bbf0b2d49">

Reply:
`NonText` is also too bright in the default colorscheme - just change it in your colorscheme imo, this PR is mostly a bugfix for #24068